packages/flutter/lib/src/widgets/context_menu_button_item.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'framework.dart';
+
+/// The buttons that can appear in a context menu by default.
+enum ContextMenuButtonType {
+  /// A button that cuts the current text selection.
+  cut,
+
+  /// A button that copies the current text selection.
+  copy,
+
+  /// A button that pastes the clipboard contents into the focused text field.
+  paste,
+
+  /// A button that selects all the contents of the focused text field.
+  selectAll,
+
+  /// Anything other than the default button types.
+  custom,
+}
+
+/// The type and callback for a context menu button.
+///
+/// See also:
+///
+///  * [DefaultTextSelectionToolbar], which can take a list of
+///    ContextMenuButtonItems and create a platform-specific context menu with
+///    the indicated buttons.
+@immutable
+class ContextMenuButtonItem {
```

Comment:
@gspencergoog I mentioned this in my PR description, but this is how I'm defining the data that can be used to generate buttons in my menu.  Otherwise everything else is just widgets.  It seems to me that something like [MenuItem](https://master-api.flutter.dev/flutter/widgets/MenuItem-class.html) wouldn't work here, but let me know if you have any ideas here or elsewhere for how to make this work better with your cascading menu work.

Reply:
Leaving a link to my answer to this question in order to make this easier to follow: https://github.com/flutter/flutter/pull/107193#issuecomment-1221245442